api/controllers/MerchStoreController.ts

```diff
@@ -158,14 +159,27 @@ export class MerchStoreController {
 
   @UseBefore(UserAuthentication)
   @Post('/item/picture/:uuid')
-  async updateMerchPhoto(@UploadedFile('image',
+  async createMerchItemPhoto(@UploadedFile('image',
     { options: StorageService.getFileOptions(MediaType.MERCH_PHOTO) }) file: File,
     @Params() params: UuidParam,
-    @AuthenticatedUser() user: UserModel): Promise<UpdateMerchPhotoResponse> {
+    @AuthenticatedUser() user: UserModel): Promise<CreateMerchPhotoResponse> {
     if (!PermissionsService.canEditMerchStore(user)) throw new ForbiddenError();
-    const picture = await this.storageService.upload(file, MediaType.MERCH_PHOTO, params.uuid);
-    const item = await this.merchStoreService.editItem(params.uuid, { picture });
-    return { error: null, item };
+
+    // generate a random string for the picture url
+    const randomID = StorageService.getRandomString();
+    const picture = await this.storageService.uploadToFolder(file, MediaType.MERCH_PHOTO, randomID, params.uuid);
+    const photo = await this.merchStoreService.createItemPhoto(params.uuid, { picture });
```

Comment:
nit: it's unclear by the variable naming what the difference is supposed to be between `picture` and `photo` - semantically they both mean essentially the same thing. How about something like `uploadedPhoto` for the former and `merchPhoto` for the latter?

Reply:
Sounds good.